### PR TITLE
Configure changelog generator

### PR DESCRIPTION
**Pull request by ccremer**
#### What this PR does / why we need it:

* Adds a changelog generator as inspired by https://github.com/ccremer/charts/releases
* Updates README

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,79 +2,9 @@
 
 The APPUiO Charts project accepts contributions via GitHub pull requests. This document outlines the process to help get your contribution accepted.
 
-## Sign Your Work
```
Comment (simu) [inline CONTRIBUTING.md:5]: Why are we removing this section?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -26,4 +26,4 @@ jobs:
           charts_dir: appuio
           charts_repo_url: https://charts.appuio.ch
         env:
-          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          CR_TOKEN: "${{ secrets.RELEASE_PAT }}"
```
Comment (simu) [inline .github/workflows/release.yml:29]: What's this secret? Doesn't seem to be configured in the repo

**Review by simu — APPROVED** [review]
LGTM overall.

File: .github/workflows/release.yml (hunk 2 above)
Comment (ccremer) [reply .github/workflows/release.yml:29]: You're right. I first need someone who has the vshnbot credentials :)
The GitHub token is insufficient as it doesn't trigger the changelog workflow ("recursive" workflow)

File: CONTRIBUTING.md (hunk 1 above)
Comment (ccremer) [reply CONTRIBUTING.md:5]: We discussed a few months ago that we don't require DCO anymore, so I removed it for cleanup reasons.

**Review by bastjan — APPROVED** [review]
I'm happy that we're removing the DCO.

**Review by glrf — APPROVED** [review]

**Pull request merged**